alembic/versions/a5c06aa9ef30_add_runmodel.py

```diff
@@ -0,0 +1,621 @@
+    print(f"All Koji builds: {all_koji_builds}")
+    print(
+        f"Koji builds deleted for no SRPM for trigger: {deleted_koji_builds_for_no_srpm}"
+    )
+    print(f"Koji builds deleted for no SRPM set: {koji_builds_without_srpm}")
+    print(
+        f"Fixed SRPM matching to trigger model from Koji build: "
+        f"{fixed_srpm_matching_from_koji_build}"
+    )
+    print("================================")
+    print(f"All Test runs: {all_test_runs}")
+    print(f"Attached correctly to build: {test_runs_attached}")
+    print(f"All Run models: {all_run_models}")
+    print(
+        f"Run models with different number of tests and builds: {number_of_builds_and_tests_differ}"
+    )
+    print(f"Run models with test run correctly set: {run_models_successful}")
+    print("================================")
+
+    # Check:
+
+    for srpm_build in session.query(SRPMBuildModel).all():
+        if not srpm_build.runs:
+            raise PackitException(
+                f"SRPMBuildModel is not connected to any RunModel: {srpm_build}"
+            )
+
+    for copr_build in session.query(CoprBuildModel).all():
+        srpm_builds = {run.srpm_build for run in copr_build.runs}
+        if len(srpm_builds) != 1:
+            raise PackitException(
+                f"More SRPM builds for one copr_build {copr_build}:\n{srpm_builds}"
+            )
+
+    for koji_build in session.query(KojiBuildModel).all():
+        srpm_builds = {run.srpm_build for run in koji_build.runs}
+        if len(srpm_builds) != 1:
+            raise PackitException(
+                f"More SRPM builds for one koji_build {koji_build}:\n{srpm_builds}"
+            )
+
+    run_model_count = 0
+    for run_model in session.query(RunModel).all():
+        run_model_count += 1
+        if not run_model.srpm_build:
+            raise PackitException(
+                f"Run model does not have SRPM build set: {run_model}"
+            )
+
+    session.commit()
+
+    # Remove direct connections:
+
+    op.drop_constraint(
+        "copr_builds_job_trigger_id_fkey", "copr_builds", type_="foreignkey"
+    )
+    op.drop_constraint(
+        "copr_builds_srpm_build_id_fkey1", "copr_builds", type_="foreignkey"
+    )
+    op.drop_column("copr_builds", "job_trigger_id")
+    op.drop_column("copr_builds", "srpm_build_id")
+    op.drop_constraint(
+        "koji_builds_srpm_build_id_fkey", "koji_builds", type_="foreignkey"
+    )
+    op.drop_constraint(
+        "koji_builds_job_trigger_id_fkey", "koji_builds", type_="foreignkey"
+    )
+    op.drop_column("koji_builds", "job_trigger_id")
+    op.drop_column("koji_builds", "srpm_build_id")
+    op.drop_constraint(
+        "srpm_builds_job_trigger_id_fkey", "srpm_builds", type_="foreignkey"
+    )
+    op.drop_column("srpm_builds", "job_trigger_id")
+    op.drop_constraint(
+        "tft_test_runs_job_trigger_id_fkey", "tft_test_runs", type_="foreignkey"
+    )
+    op.drop_column("tft_test_runs", "job_trigger_id")
```

Comment:
is there a risk that the DB would become inconsistent if something goes wrong during runing these?

Reply:
TBH, I am not sure if there is some transaction going on during this. I would say it shouldn't if it works for the current state of DB. But I can test that on the real data right before the deployment.

Reply:
I can try raising an error during those steps locally and see what happens. (And maybe move the `commit` call to the very end of this function.)

Reply:
I can confirm now that in the case of `Exception` the changes are not applied.